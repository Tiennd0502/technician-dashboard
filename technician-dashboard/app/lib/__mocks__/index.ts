export * from './table';
export * from './categories';
export * from './services';
