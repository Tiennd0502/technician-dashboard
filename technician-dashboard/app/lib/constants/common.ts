export const LOGO_URL = '/images/technician.svg';

export const AVATAR_DEFAULT = '/images/empty-avatar.svg';

export const CURRENCY_CHARACTERS = {
  EURO: 'â‚¬',
};

export const TRUNCATE_STYLE = {
  textOverflow: 'ellipsis',
  overflow: 'hidden',
  whiteSpace: 'nowrap',
};

export const NO_DATA = 'No data found';
