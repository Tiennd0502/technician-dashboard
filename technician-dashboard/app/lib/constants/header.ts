import { AVATAR_DEFAULT } from '.';

export const HEADER_INFO = {
  user: {
    name: 'Belle',
    avatar: AVATAR_DEFAULT,
  },
  title: 'Product & Services',
};
