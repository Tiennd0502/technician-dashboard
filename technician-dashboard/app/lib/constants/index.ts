export * from './common';
export * from './filter';
export * from './sidebar';
export * from './header';
export * from './rules';
export * from './messages';
export * from './routes';
export * from './status';
export * from './table';
export * from './categories';
export * from './services';
