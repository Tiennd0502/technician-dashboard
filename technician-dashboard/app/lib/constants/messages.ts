export const ERROR_MESSAGE = {
  FIELD_REQUIRED: (fieldName: string) => `${fieldName} is required.`,
};
