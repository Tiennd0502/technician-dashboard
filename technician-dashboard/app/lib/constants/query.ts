export const QUERY_KEYS = {
  CATEGORIES: ['categories'],
  SERVICES: ['services'],
};
