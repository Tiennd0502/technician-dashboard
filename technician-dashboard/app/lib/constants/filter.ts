import { SORT_TYPE } from '@/lib/interfaces';

export const PRODUCT_LIMIT = 5;

export const DEFAULT_PRODUCT_FILTER = {
  name: '',
  sortBy: 'name',
  order: SORT_TYPE.Asc,
};
