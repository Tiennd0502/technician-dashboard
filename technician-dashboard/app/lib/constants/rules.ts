import { ERROR_MESSAGE } from './messages';

export const PRODUCT_RULES = {
  name: {
    required: ERROR_MESSAGE.FIELD_REQUIRED('Product Name'),
  },
  brand: {
    required: ERROR_MESSAGE.FIELD_REQUIRED('Brand Name'),
  },
  service: {
    required: ERROR_MESSAGE.FIELD_REQUIRED('Service Name'),
  },
};
