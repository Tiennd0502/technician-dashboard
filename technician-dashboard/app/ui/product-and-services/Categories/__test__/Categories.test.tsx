import { render } from '@testing-library/react';

import { CATEGORIES } from '@/lib/constants';
import Categories from '..';

describe('Categories test cases', () => {
  const props = {
    list: CATEGORIES,
  };

  test('should render correctly', () => {
    const { container } = render(<Categories {...props} />);
    expect(container).toMatchSnapshot();
  });

  test('Should render enough elements', () => {
    const { getByTestId } = render(<Categories {...props} />);
    const categories = getByTestId('list-category');

    expect(categories.childNodes).toHaveLength(CATEGORIES.length);
  });

  test('Should render empty elements', () => {
    const { getByTestId } = render(<Categories list={[]} />);
    const categories = getByTestId('list-category');

    expect(categories.childNodes).toHaveLength(0);
  });
});
