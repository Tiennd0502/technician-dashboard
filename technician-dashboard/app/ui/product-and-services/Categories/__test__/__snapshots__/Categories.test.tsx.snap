// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories test cases should render correctly 1`] = `
<div>
  <div
    class="css-1j52cd2"
    role="list"
  >
    <a
      class="chakra-link css-mis0aa"
      href="#"
    >
      <p
        class="chakra-text css-1kmnq3w"
      >
        Sofa Cleaning
      </p>
    </a>
    <a
      class="chakra-link css-kysffd"
      href="#"
    >
      <p
        class="chakra-text css-1kmnq3w"
      >
        Bathroom
      </p>
    </a>
    <a
      class="chakra-link css-f8l8lm"
      href="#"
    >
      <p
        class="chakra-text css-1kmnq3w"
      >
        Best Control
      </p>
    </a>
    <a
      class="chakra-link css-11nnblt"
      href="#"
    >
      <p
        class="chakra-text css-1kmnq3w"
      >
        View All
      </p>
    </a>
  </div>
</div>
`;
